.github/workflows/reusable-solidity-docs.yml

```diff
@@ -0,0 +1,248 @@
+name: Generate and push Solidity API documentation
+
+# To use this workflow, make sure `solidity-docgen` package is added to the 
+# projects's dev dependencies and is configured in Hardhat config (see
+# https://github.com/OpenZeppelin/solidity-docgen#usage).
+# The workflow expects `outputDir` to be set to `generated-docs`. You may
+# also need to specify other configs, like `temlates` or `exclude`.
+on:
+  workflow_call:
+    inputs:
+      projectSubfolder:
+        description: "A root's subfolder containing the `contracts` folder and
+          the Yarn / Hardhat configuration. Leave empty ('') if the project
+          config resides in the root. If config resides in a subfolder, provide
+          the subfolder name/path, without leading `.` and ending `/`. For
+          example, type `/v2/solidity` if the path to the `contracts` folder is
+          `./v2/solidity/contracts`."
+        type: string
+        required: false
+        default: ""
+      removeTrailingSpacesInComments:
```

Comment:
This should be `removeLeadingSpacesInComments` or maybe we could simplify it to just `trim` or `trimComments`?

Reply:
See 283d7a77c0b64b1978f6e6aae3f0a8d1e6553a7b.